src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,72 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React, { useEffect, useState } from 'react';
+import { Filter, FilterField } from '../../types/Filter';
+
+type Props = {
+  onFilter: (filter: Filter) => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({ onFilter }) => {
+  const [query, setQuery] = useState('');
+  const [filterField, setFilterField] = useState<FilterField>(FilterField.All);
+
+  useEffect(() => {
+    onFilter({ filterField, query });
+  }, [query, filterField]);
+
+  const handleChangeFilterField
+  = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setFilterField(event.target.value as FilterField);
+  };
+
+  const handleChangeQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const handleDeleteQuery = () => {
+    setQuery('');
+  };
```

Comment:
You can call `onFilter` function inside these handlers, and avoid using `useEffect`